Quantized conv avoid functional usage

z-a-f opened the pull request:
Stack:
&nbsp;&nbsp;&nbsp;&nbsp;:black_circle:&nbsp; **#21808 Quantized conv avoid functional usage**&nbsp;&nbsp;[:yellow_heart:](https://our.internmc.facebook.com/intern/diff/D15835572/)
&nbsp;&nbsp;&nbsp;&nbsp;:white_circle:&nbsp; #21323 Quantized Conv2d Module&nbsp;&nbsp;[:yellow_heart:](https://our.internmc.facebook.com/intern/diff/D15551835/)

This refactor changes the conv module to avoid the usage of the functional ops.

Differential Revision: [D15835572](https://our.internmc.facebook.com/intern/diff/D15835572/)

raghuramank100 commented on torch/nn/quantized/functional.py:
```diff
@@ -21,7 +17,7 @@ def conv2d(input, weight, bias,
            scale=1.0, zero_point=0,
            dtype=torch.quint8,
-           prepacked=True):
+           prepacked=False):
```
Lets remove prepacked as an option, functionals by default should always do prepacking.

raghuramank100 commented on torch/nn/quantized/modules/conv.py:
```diff
@@ -38,14 +39,31 @@ def __init__(self, weight, bias, scale, zero_point, dtype,
+    def weight(self):
+        return self._packed_weight
+
+    @weight.setter
```
Great practice!. Will use this for other modules too.

raghuramank100 commented on torch/nn/quantized/modules/conv.py:
```diff
@@ -38,14 +39,31 @@ def __init__(self, weight, bias, scale, zero_point, dtype,
-        self.weight = weight
-        self._packed_weight = ops.quantized.fbgemm_conv_prepack(weight, groups)
-        self.bias = bias  # torch.nn.Parameter(bias, requires_grad=False)
+        self._weight = weight  # Store the original weight for future reference.
```
Please remove _weight. Quantized modules should be smaller in size and we should not store multiple copies of weights. Let us generate the unpacked weights when it is needed.

raghuramank100 reviewed (changes requested):
Please address requested changes, thanks

raghuramank100 reviewed (changes requested):
Please address requested change, thanks

jerryzh168 commented on torch/nn/quantized/modules/conv.py:
```diff
@@ -79,15 +94,10 @@ def __init__(self, weight, bias, scale, zero_point, dtype,
-                         prepacked=True)
+        if input.ndim != 4:
+            raise ValueError("Input shape must be `(N, C, H, W)`!")
+        return ops.quantized.fbgemm_conv2d(input,
```
why do we avoid functional usage?

zafartahirov replied:
That was a request, because we want the functional to always prepack the weights, while the module to prepack only in the constructor

jerryzh168 replied:
you mean functional will take a quantized weight rather than prepacked and quantized weight as input?

zafartahirov replied:
Correct. We need to document it so that the user would be aware that the functional is less efficient :/

jerryzh168 commented on torch/nn/quantized/modules/conv.py:
```diff
@@ -38,14 +39,27 @@ def __init__(self, weight, bias, scale, zero_point, dtype,
+        self._packed_weight = ops.quantized.fbgemm_conv_prepack(weight,
+                                                                self.groups)
+        self.bias = bias
+        self.bias.requires_grad = False  # Inference only!
```
I think users will notice when they run backward? I feel just change bias to requires_grad=False does not prevent them to run backward since they will still hit some other tensor that don't have this set to False?

zafartahirov replied:
Yea, this was a temp hack. Cleaned it up :)

bwasti committed:
Cleanup integer sign warnings

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/22560

Test Plan: Imported from OSS

Differential Revision: D16151479

Pulled By: bwasti

fbshipit-source-id: a54139d8f95ed964530862f96723e4365724b2da

Commit:
README for the quantized op creation (#22165)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/22165

Workflow description for the quantized ops design.

Reviewed By: jerryzh168

Differential Revision: D15975977

fbshipit-source-id: ef73b172f609adef149c157c404bb452b5457a9f

Commit:
Report errors through call stack (#22280)

Summary:
The error for `test_error_stack_module`:

```
Traceback (most recent call last):
  File "../test.py", line 35, in <module>
    scripted = torch.jit.script(M())
  File "/home/davidriazati/other/pytorch/torch/jit/__init__.py", line 1119, in script
    return _convert_to_script_module(obj)
  File "/home/davidriazati/other/pytorch/torch/jit/__init__.py", line 1825, in _convert_to_script_module
    raise e
RuntimeError:

d(int x) -> int:
Expected a value of type 'int' for argument 'x' but instead found type 'str'.
:
at ../test.py:11:12
def c(x):
    return d("hello") + d(x)
           ~ <--- HERE

'c' is being compiled since it was called from 'b'
at ../test.py:14:12
def b(x):
    return c(x)
           ~~~ <--- HERE

'b' is being compiled since it was called from 'forward'
at ../test.py:22:16
    def forward(self, x):
        return b(x)
               ~~~ <--- HERE

'forward' is being compiled since it was called from 'forward'
at ../test.py:31:20
    def forward(self, x):
        return x + self.submodule(x)
                   ~~~~~~~~~~~~~~~~ <--- HERE
```

This also unifies our error reporting in the front end with `ErrorReport`

TODO
* Include module names in message, #22207 should make this easy

](https://our.intern.facebook.com/intern/diff/16060781/)
Pull Request resolved: https://github.com/pytorch/pytorch/pull/22280

Pulled By: driazati

Differential Revision: D16060781

fbshipit-source-id: c42968b53aaddb774ac69d5abbf7e60c23df8eed

jerryzh168 committed:
fix lint (#22654)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/22654

att

Reviewed By: bddppq

Differential Revision: D16168219

fbshipit-source-id: db1a5e2161e7be70b2f6e6b4beaa27ea91f853f2

mingzhe09088 committed:
introduce a new interface to add a list of operators (#21209)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/21209

This diff introduces a new interface to add a list of operators. Here are the steps to add ops using this interface:

- create op_list:
```unary_ops_list = op_bench.op_list(
    attr_names=["op_name", "op_function"],
    attrs=[
         ["abs", torch.abs],
         ["abs_", torch.abs_],
   ],
)
```
-  create a bench class:
```
class UnaryOpBenchmark(op_bench.TorchBenchmarkBase):
    def init(self, M, N, op_function):
        self.input_one = torch.rand(M, N)
        self.op_func = op_function

    def forward(self):
        return self.op_func(self.input_one)
```
- 3. register those ops
``` op_bench.generate_pt_tests_from_list(unary_ops_list, unary_ops_configs, UnaryOpBenchmark)
 ```

Reviewed By: zheng-xq

Differential Revision: D15514188

fbshipit-source-id: f09b359cab8175eeb8d51b3ad7bbbcfbc9f6430f

mingzhe09088 committed:
change unary, pool, max ops to use new interface (#22661)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/22661

as title

Reviewed By: hl475

Differential Revision: D16170825

fbshipit-source-id: d80944224b8717e7aa35980907ff48e587b85217

ssnl committed:
Avoid potential extra copy in _lu_with_info_cuda (#22634)

Summary:
No need to `clone` if the expanded size matches original size.
Pull Request resolved: https://github.com/pytorch/pytorch/pull/22634

Differential Revision: D16171091

Pulled By: ezyang

fbshipit-source-id: 3d8f116398f02952488e321c0ee0ff2868768a0c

Krovatkin committed:
Add traces to LowerGradOf and SpecializeAutoGrad

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/22599

Differential Revision: D16161144

Pulled By: Krovatkin

fbshipit-source-id: 9e206fcfb1796e9448e80f178b75d0c277bd348f

andyljones committed:
Properly formats errors rising up from C++ extension compilation (#22445)

Summary:
Here's a C++ extension with a missing semicolon:
```python
torch.utils.cpp_extension.load_inline('test', 'int main() { return 0 }')
```
which currently generates this error
```
RuntimeError: Error building extension 'test_v6': b'[1/2] c++ -MMD -MF main.o.d -
DTORCH_EXTENSION_NAME=test_v6 -DTORCH_API_INCLUDE_EXTENSION_H -isystem
/opt/conda/lib/python3.7/site-packages/torch/include -isystem /opt/conda/lib/python3.7/site-
packages/torch/include/torch/csrc/api/include -isystem /opt/conda/lib/python3.7/site-
packages/torch/include/TH -isystem /opt/conda/lib/python3.7/site-packages/torch/include/THC
-isystem /opt/conda/include/python3.7m -D_GLIBCXX_USE_CXX11_ABI=0 -fPIC -std=c++11 -c
/tmp/torch_extensions/test/main.cpp -o main.o\nFAILED: main.o \nc++ -MMD -MF main.o.d -
DTORCH_EXTENSION_NAME=test_v6 -DTORCH_API_INCLUDE_EXTENSION_H -isystem
/opt/conda/lib/python3.7/site-packages/torch/include -isystem /opt/conda/lib/python3.7/site-
packages/torch/include/torch/csrc/api/include -isystem /opt/conda/lib/python3.7/site-
packages/torch/include/TH -isystem /opt/conda/lib/python3.7/site-packages/torch/include/THC
 -isystem /opt/conda/include/python3.7m -D_GLIBCXX_USE_CXX11_ABI=0 -fPIC -std=c++11 -c
/tmp/torch_extensions/test/main.cpp -o main.o\n/tmp/torch_extensions/test/main.cpp: In
function \xe2\x80\x98int main()\xe2\x80\x99:\n/tmp/torch_extensions/test/main.cpp:2:23:
error: expected \xe2\x80\x98;\xe2\x80\x99 before \xe2\x80\x98}\xe2\x80\x99 token\n int
main() { return 0 }\n                       ^\nninja: build stopped: subcommand failed.\n'
```

After this PR, the error is
```
RuntimeError: Error building extension 'test': [1/2] c++ -MMD -MF main.o.d -
DTORCH_EXTENSION_NAME=test -DTORCH_API_INCLUDE_EXTENSION_H -isystem
/opt/conda/lib/python3.7/site-packages/torch/include -isystem /opt/conda/lib/python3.7/site-
packages/torch/include/torch/csrc/api/include -isystem /opt/conda/lib/python3.7/site-
packages/torch/include/TH -isystem /opt/conda/lib/python3.7/site-packages/torch/include/THC
 -isystem /opt/conda/include/python3.7m -D_GLIBCXX_USE_CXX11_ABI=0 -fPIC -std=c++11 -c
/tmp/torch_extensions/test/main.cpp -o main.o
FAILED: main.o
c++ -MMD -MF main.o.d -DTORCH_EXTENSION_NAME=test -
DTORCH_API_INCLUDE_EXTENSION_H -isystem /opt/conda/lib/python3.7/site-
packages/torch/include -isystem /opt/conda/lib/python3.7/site-
packages/torch/include/torch/csrc/api/include -isystem /opt/conda/lib/python3.7/site-
packages/torch/include/TH -isystem /opt/conda/lib/python3.7/site-packages/torch/include/THC
 -isystem /opt/conda/include/python3.7m -D_GLIBCXX_USE_CXX11_ABI=0 -fPIC -std=c++11 -c
/tmp/torch_extensions/test/main.cpp -o main.o
/tmp/torch_extensions/test/main.cpp: In function ‘int main()’:
/tmp/torch_extensions/test/main.cpp:2:23: error: expected ‘;’ before ‘}’ token
 int main() { return 0 }
                       ^
ninja: build stopped: subcommand failed.
```
which is a lot easier to read.
Pull Request resolved: https://github.com/pytorch/pytorch/pull/22445

Differential Revision: D16094205

Pulled By: ezyang

fbshipit-source-id: 21043344aac260dc3e4e04d6a42898507bb840e4

xuhdev committed:
Add comments about adding new build options. (#22641)

Summary:
Also revert the change of cmake.py in
c97829d7011bd59d662f6af9c3a0ec302e7e75fc . The comments are added to
prevent future similar incidents in the future (which has occurred a couple of times in the past).
Pull Request resolved: https://github.com/pytorch/pytorch/pull/22641

Differential Revision: D16171763

Pulled By: ezyang

fbshipit-source-id: 5a65f9fbb3c1c798ebd25521932bfde0ad3d16fc

mingzhe09088 committed:
Enable C2 operators running with {cpu, gpu} * {forward, backward} (#22664)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/22664

This diff enables c2 operators to run the combination of {cpu, gpu} * {forward, backward}.

Reviewed By: hl475

Differential Revision: D15781789

fbshipit-source-id: e9843e3c46ea144042829860638d406f6a33792b

mingzhe09088 committed:
change c2 bench to use new tensor creation interface (#22663)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/22663

as title

Reviewed By: hl475

Differential Revision: D15744502

fbshipit-source-id: 441ab9fb7580ca87c3f2027d0a63ba18b8d35016

xzhu1900 committed:
add forward declaration in stateful dataset (#22562)

Summary:
Addressing potential dependency issue by adding forward declaration for OutputArchive/InputArchive.

This change follows the same pattern in base.h in 'torch/csrc/api/include/torch/data/samplers/base.h'
Pull Request resolved: https://github.com/pytorch/pytorch/pull/22562

Differential Revision: D16161524

Pulled By: soumith

fbshipit-source-id: d03f8a2ece5629762f9fa8a27b15b0d037e8f07b

izdeby committed:
Intoducing bfloat16 type (#21522)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/21522
ghimport-source-id: 4803f197ec04938501fdb10c1741280331c349d2

Test Plan: Imported from OSS

Differential Revision: D15819369

Pulled By: izdeby

fbshipit-source-id: 46408dc316a5c4dc644a736dc42da2422b34bcb9

Commit:
Updating submodules

Reviewed By: zpao

fbshipit-source-id: f5636ab0457c1b2e15df95a5677a7194978d9cd0

izdeby committed:
Enabled BFloat16 storage (#21523)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/21523
ghimport-source-id: 698b3cbd6b21c09b9ff8bf8011980df8e35c33b0

Test Plan: Imported from OSS

Differential Revision: D15819368

Pulled By: izdeby

fbshipit-source-id: f6b3bba7b3ca8ee677bd80a231dbb3920c07d61c

ezyang committed:
Revert D16161144: [pytorch][PR] Add traces to LowerGradOf and SpecializeAutoGrad

Differential Revision:
D16161144

Original commit changeset: 9e206fcfb179

fbshipit-source-id: 8f9eecb5cd6ca715bd0c647c32cf77cd9d88e6ac

tianzhi0549 committed:
keep reuqires_grad unchanged after converting bn to syncbn (#22569)

Summary:
After converting BN layers to SyncBN layers, the function will set all `requires_grad = True` regardless of the original requires_grad states. I think it is a bug and have fixed it in this PR.
Pull Request resolved: https://github.com/pytorch/pytorch/pull/22569

Differential Revision: D16151647

Pulled By: zou3519

fbshipit-source-id: e2ad1886c94d8882485e7fb8be51ad76469ecc67

izdeby committed:
Added Bfloat16 tensor for cpu with very limited support (#21860)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/21860
ghimport-source-id: 5290755b63033cdfdeb911a4ecf4aa282b3db02d

Test Plan: Imported from OSS

Differential Revision: D15856091

Pulled By: izdeby

fbshipit-source-id: 54e7e17be1b5c5a2e80a41feaeaeba75dbb8108f

Krovatkin committed:
Enable more passes in ProfilingGraphExecutor

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/22079

Differential Revision: D16119322

Pulled By: Krovatkin

fbshipit-source-id: 301fcc42d0e1f031d9de5bcd9679fb8c2d742fef

jerryzh168 committed:
fix print (#22689)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/22689

att

Reviewed By: Lucaskabela

Differential Revision: D16184260

fbshipit-source-id: 1a6ad51a37918d0c81d6e3baa0ca0baa32cb9673

xw285cornell committed:
skip import nccl and gloo_gpu in cpu machine (#22522)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/22522

Skip importing nccl and gloo_gpu modules in cpu machine

Reviewed By: bddppq

Differential Revision: D16115827

fbshipit-source-id: 329b7a0bb5eccb78c9e772bdab5db7c79b546d55

suo committed:
fix dce over loops

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/22632

Test Plan: Imported from OSS

Differential Revision: D16184469

Pulled By: suo

fbshipit-source-id: b7cc2d20a7dd8b287e1b6128ddb70d3936032a7e

Commit:
Refactoring _wrap_outputs to remove python dependence.

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/22631

Test Plan:
test suite

Imported from OSS

Differential Revision: D16185040

fbshipit-source-id: 9b83749f6c9cd05d13f54a3bb4801e263293252b

xuhdev committed:
Add a bitwise NOT operator for integer and Boolean types (CUDA).

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/22320

Test Plan: Imported from OSS

Differential Revision: D16183578

Pulled By: colesbury

fbshipit-source-id: 2f72cce5e10fd637be1ac87e1bbfe0937a661034

xuhdev committed:
Delegate Python ~ (invert operator) to Tensor.bitwise_not().

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/22326

Test Plan: Imported from OSS

Differential Revision: D16183577

Pulled By: colesbury

fbshipit-source-id: f86838c407db4ded9ce70998bf1ab1ffd75b3b58

xuhdev committed:
Add a bitwise NOT operator for integer and Boolean types (CPU).

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/22283

Test Plan: Imported from OSS

Differential Revision: D16183576

Pulled By: colesbury

fbshipit-source-id: 2e539fab8ff885dddb9bff334d1d784b28d65b8f

xuhdev committed:
Support Half type in randperm.

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/22102

Test Plan: Imported from OSS

Differential Revision: D16153586

Pulled By: li-roy

fbshipit-source-id: d58e3dbc5da893005f4eaf521a28b0d752274eff

xuhdev committed:
Swap detection order in randperm_out_cuda to avoid unnecessary conversion from float when the input is small.

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/22103

Test Plan: Imported from OSS

Differential Revision: D16153585

Pulled By: li-roy

fbshipit-source-id: 0801b91e7b352c8de8fdfbe929be85d69182b8da

xuhdev committed:
Refactor and improve randperm tests.

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/22121

Test Plan: Imported from OSS

Differential Revision: D16153794

Pulled By: li-roy

fbshipit-source-id: 4dbfa6cfcc79f6d431918a6646664215fa9ea0b9

Commit:
Add common_quantized test case utilities (#22694)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/22694

Move quantization and quantized utility functions for testing to common_quantized.py and common_quantization.py.  Addditionally, add a quantized test case base class which contains common methods for checking the results of quantization on modules.  As a consequence of the move, fixed the import at the top of test_quantized.py, and test_quantization to use the new utility

Reviewed By: jerryzh168

Differential Revision: D16172012

fbshipit-source-id: 329166af5555fc829f26bf1383d682c25c01a7d9

Commit:
Fix Trace inlining of graphs with optional inputs (#22686)

Summary:
Previously in tracing when we called a script function we would inline the graph and set the graph inputs equal to the types the graph was invoked with.

This breaks for optional arguments invoked with None since we rely on None being set to Optional[T] in schema matching.
Pull Request resolved: https://github.com/pytorch/pytorch/pull/22686

Differential Revision: D16186372

Pulled By: eellison

fbshipit-source-id: e25c807c63527bf442eb8b31122d50689c7822f5

ssnl committed:
Fix a FIXME in test_nn (#22675)

Summary:
https://github.com/pytorch/pytorch/issues/17262 is already resolved, so this should pass now.
Pull Request resolved: https://github.com/pytorch/pytorch/pull/22675

Differential Revision: D16188003

Pulled By: zou3519

fbshipit-source-id: 32693229a0590b274ed1bf76b815f17e77c2d3ea

wanchaol committed:
register __getitem__ builtin

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/22276

Test Plan: Imported from OSS

Differential Revision: D16060595

Pulled By: wanchaol

fbshipit-source-id: e1e27d6be8d62fc1a841860a783aff108980d9d3

ezyang committed:
Turn on USE_DIRECT_NVRTC in fbcode again. (#22685)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/22685
ghstack-source-id: 86247780

Reviewed By: bddppq

Differential Revision: D16182352

fbshipit-source-id: fc51aa7c1112904b8cccd055dc87e10c836cf2fb

Commit:
Move Quantization Models to common_quantization (#22706)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/22706

Moved the models used for quantization test from the test_quantization.py file to common_quantization.py

Reviewed By: jerryzh168

Differential Revision: D16189865

fbshipit-source-id: 409b43454b6b3fe278ac16b1affb9085d6ed6835

suo committed:
Give functions qualified names (#22206)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/22206
ghimport-source-id: d453219d907e048f24eb7f63c096b2c300307c83

Test Plan: Imported from OSS

Differential Revision: D15998762

Pulled By: suo

fbshipit-source-id: bc2b734f626ab07f97dc50ddf1b021e8b46de312

suo committed:
make CompilationUnit::define return defined functions

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/22667

Test Plan: Imported from OSS

Differential Revision: D16184799

Pulled By: suo

fbshipit-source-id: 9f77a7ca2223237fbcb4b12a4734b7d334f7be13

suo committed:
refactor self to be a class again (#22207)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/22207
ghimport-source-id: 36ee8bd17411a2e220665ad2a27364653061070e

Test Plan: Imported from OSS

Differential Revision: D15998758

Pulled By: suo

fbshipit-source-id: 14bad87bb6e44bf1a43ae86339d8cc7b311c76dd

Krovatkin committed:
add line numbers to jit_log.h

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/22630

Differential Revision: D16172090

Pulled By: Krovatkin

fbshipit-source-id: 26cdb0077a0bfbf9981e39359472f3251546db53

nairbv committed:
add a comment to native_functions explaining softmax interfaces (#22651)

Summary:
Address the review comment made by gchanan here:

https://github.com/pytorch/pytorch/pull/22456#discussion_r300715866
Pull Request resolved: https://github.com/pytorch/pytorch/pull/22651

Differential Revision: D16181828

Pulled By: nairbv

fbshipit-source-id: 0d41a9024c2664298c281e198a997be73e7f8499

izdeby committed:
fixing lint

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/22703

Differential Revision: D16188326

Pulled By: izdeby

fbshipit-source-id: 72e6b6f957068c3995010a1b811f24cd2304ff6f

Commit:
Revert D15998762: [jit] Give functions qualified names

Differential Revision:
D15998762

Original commit changeset: bc2b734f626a

fbshipit-source-id: a118cc4e9a34233279e8380529a8d8120a25839d

Commit:
Revert D15998758: [jit] refactor self to be a class again

Differential Revision:
D15998758

Original commit changeset: 14bad87bb6e4

fbshipit-source-id: f2c29974d4afc4d8f88a36e9c266e6d5a22a6191

Commit:
Revert D16184799: [jit] make CompilationUnit::define return defined functions

Differential Revision:
D16184799

Original commit changeset: 9f77a7ca2223

fbshipit-source-id: a0e08220d924a6ca55bf2f1f77754553d0133595

colesbury committed:
Revert D16183577: Delegate Python ~ (invert operator) to Tensor.bitwise_not().

Differential Revision:
D16183577

Original commit changeset: f86838c407db

fbshipit-source-id: bbf53ce52a20b1e90b1fe522d73e558d8044c4ba

mrshenli committed:
Mark Unpickler data ptr arg as const

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/22690

Differential Revision: D16184299

Pulled By: mrshenli

fbshipit-source-id: 332954028533952dad01df03eca8e95bf6fe67a9

Commit:
Quantized Conv2d Module (#21323)

Summary:
Stack:
&nbsp;&nbsp;&nbsp;&nbsp;:white_circle:&nbsp; https://github.com/pytorch/pytorch/issues/21808 Quantized conv avoid functional usage&nbsp;&nbsp;[:yellow_heart:](https://our.intern.facebook.com/intern/diff/D15835572/)
&nbsp;&nbsp;&nbsp;&nbsp;:black_circle:&nbsp; **https://github.com/pytorch/pytorch/issues/21323 Quantized Conv2d Module**&nbsp;&nbsp;[:yellow_heart:](https://our.intern.facebook.com/intern/diff/D15551835/)

Quantized Conv2d Module
Pull Request resolved: https://github.com/pytorch/pytorch/pull/21323

Test Plan:
Tests are split into two parts: functional and API.

`buck test mode/dev caffe2/test:quantized -- test_conv_api` : https://our.intern.facebook.com/intern/testinfra/testrun/4785074605318491

```
Parsing buck files: finished in 1.4 sec
Building: finished in 4.6 sec (100%) 7136/7136 jobs, 2 updated
  Total time: 6.1 sec
Trace available for this run at /tmp/testpilot.20190703-153023.392592.log
TestPilot test runner for Facebook. See https://fburl.com/testpilot for details.
Testpilot build revision 7149de230b9e1cdc7a872bb31fe099f0616dee09 fbpkg e59e6ab0fe8e47a496f915d34555c3ad at Fri Jun 28 12:20:54 2019 by twsvcscm from /usr/local/fbprojects/packages/testinfra.testpilot/647/t.par
Discovering tests
Running 2 tests
Started new test run: https://our.intern.facebook.com/intern/testinfra/testrun/4785074605318491
      ✓ caffe2/test:quantized - test_conv_api (test_nn_quantized.ModuleAPITest) 0.044 1/2 (passed)
      ✓ caffe2/test:quantized - test_conv_api (test_quantized_conv.FunctionalAPITest) 5.109 2/2 (passed)
Finished test run: https://our.intern.facebook.com/intern/testinfra/testrun/4785074605318491
Summary (total time 9.08s):
  PASS: 2
  FAIL: 0
  SKIP: 0
  FATAL: 0
  TIMEOUT: 0
  OMIT: 0
```

Differential Revision: D15551835

Pulled By: zafartahirov

fbshipit-source-id: 481a7df4b8a88e485437e1596eefb08d5e6766fa

Commit:
V8: Fixed a bug where we pass packed input to the conv2d function

Differential Revision: D15835572
Differential Version: 86310356

zafartahirov commented:
I have no idea what happened -- created a different one: #22733

Commit:
V9: Rebase

Differential Revision: D15835572
Differential Version: 86310782
